media-proxy/include/mesh/mesh_json_proxy.h

```diff
@@ -0,0 +1,217 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef MESH_JSON_PROXY_H
+#define MESH_JSON_PROXY_H
+
+#include "json.hpp"
+using json = nlohmann::json;
+
+namespace mesh {
+
+struct MultipointGroup {
+    std::string urn;
+};
+
+enum ST2110_Transport {
+    TRANSPORT_ST2110_INVALID,
+    TRANSPORT_ST2110_20,
+    TRANSPORT_ST2110_22,
+    TRANSPORT_ST2110_30,
+    TRANSPORT_ST2110_MAX
+};
+
+enum ST2110_Pacing {
+    PACING_ST2110_INVALID,
+    PACING_ST2110_DEFAULT,
+    PACING_ST2110_20_NARROW,
+    PACING_ST2110_20_WIDE,
+    PACING_ST2110_20_LINEAR,
+    PACING_ST2110_30_RL,
+    PACING_ST2110_30_TSC,
+    PACING_ST2110_MAX
+};
+
+struct ST2110Config {
+    ST2110_Transport transport;
+    std::string remoteIpAddr;
+    uint remotePort;
+    ST2110_Pacing pacing;
+    uint payloadType;
+};
+
+enum RDMA_ConnectionMode {
+    CONNECTION_MODE_INVALID,
+    CONNECTION_MODE_DEFAULT,
+    CONNECTION_MODE_RC,
+    CONNECTION_MODE_UC,
+    CONNECTION_MODE_UD,
+    CONNECTION_MODE_RD,
+    CONNECTION_MODE_MAX,
+};
+
+struct RdmaConfig {
+    RDMA_ConnectionMode connectionMode;
+    uint maxLatencyNs;
+};
+
+struct Connection {
+    std::vector<MultipointGroup> multipointGroup;
+    std::vector<ST2110Config> st2110;
+    std::vector<RdmaConfig> rdma;
+};
+
+enum Video_Format {
+    VIDEO_FORMAT_INVALID,
+    VIDEO_FORMAT_NV12,        ///< planar YUV 4:2:0, 12bpp
+    VIDEO_FORMAT_YUV422P,     ///< planar YUV 4:2:2, 16bpp
+    VIDEO_FORMAT_YUV422P10LE, ///< planar YUV 4:2:2, 20bpp
+    VIDEO_FORMAT_YUV444P10LE, ///< planar YUV 4:4:4, 30bpp
+    VIDEO_FORMAT_RGB8,        ///< packed RGB 3:3:2,  8bpp
```

Comment:
Do we really plan to support very lossy RGB8 (**8bpp**), or RGB24 (24bpp, **8-bit** format)?

Reply:
FFmpeg describes RGB8 as 2:3:3. 3:3:2 is a BGR8.
```
IO... bgr8                   3              8      3-3-2
```